.taskcluster.yml

```diff
@@ -1,102 +1,96 @@
-      # resources within which the task should be executed. Pools may use different
-      # EC2 instance types, AMIs (Amazon Machine Images), etc.
-      # Generally, github-worker is a variant of docker-worker in TC docs
-      workerType: github-worker # TC's generic worker for any github projects; can do Try server pushes
-      # A scope is a permission to perform a particular action. A task has a set of
-      # scopes that determine what the task can do. These scopes are validated by the worker.
-      payload: # The format of a task's payload property is specific to the worker that will execute it
-        maxRunTime: 600 # 10 minutes
-        image: "node:8" # The Docker image to pull
-        # The worker will automatically upload any artifacts upon task completion
-        artifacts: # syntax depends on the workerType
-          public/web-ext-artifacts:
-            # contentType is not a valid key for workerType github-worker
-            expires: {$fromNow: '14 days'}
-            path: /repo/web-ext-artifacts
-            type: directory
-        env: # Environment variables to declare in the shell
-          DEBUG: "* -eslint*" # Say you have debug statements in library code; a DEBUG variable with a
-          # value of * will log all levels of debug statements
-        command: # The command to run within that Docker image
-          - "/bin/bash"
-          - "-lc" # '--login' (executes user's profile commands), '-c' (executes commands from a string)
-          - >-
-            git clone ${event.pull_request.head.repo.git_url} repo &&
-            cd repo &&
-            git checkout ${event.pull_request.head.sha} &&
-            npm install &&
-            npm run build &&
-            npm run lint &&
-            npm run package &&
-            /repo/bin/commit_xpi.sh
-      metadata:
-        name: "Firefox Experiments GitHub Tests"
-        description: "All tests"
-        owner: ${event.pull_request.user.login}@users.noreply.github.com
-        source: ${event.repository.url}
-  # When a commit is pushed to master, perform this task
-  # Note: Currently this is wrongly showing a status on PR pushes in GitHub (Bug 1514914)
-  # If a branch is merged without being updated first, the tests could pass on the branch
-  # but not when things are merged into master.
-  - $if: 'tasks_for == "github-push"'
-    then:
-      $if: 'event.ref == "refs/heads/master"'
+  $let:
+    # This only supports pushes and pull requests for now. If we start responding to another
+    # action type, make this a $match.
+    repo:
+      $if: 'tasks_for == "github-push"'
       then:
-        taskId: {$eval: as_slugid("push_task")}
+        git_url: ${event.repository.url}
+        url: ${event.repository.url}
+        ref: ${event.after}
+      else:
+        git_url: ${event.pull_request.head.repo.git_url}
+        url: ${event.pull_request.head.repo.url}
+        ref: ${event.pull_request.head.sha}
+  in:
+    $let:
+      taskRef:
```

Comment:
Hmm, yeah. These are all my notes as I was learning how to create this config file and what all the pieces are doing. I'm going to kick this can down the road for now, since we have a ways to go before sharing this out more widely.